ibis/tests/all/test_vectorized_udf.py

```diff
@@ -21,15 +26,41 @@ def calc_mean(s):
 @pytest.mark.only_on_backends([Pandas, PySpark])
 @pytest.mark.xfail_unsupported
 def test_elementwise_udf(backend, alltypes, df):
-    @elementwise(input_type=[dt.double], output_type=dt.double)
-    def add_one(s):
-        return s + 1
-
     result = add_one(alltypes['double_col']).execute()
     expected = add_one.func(df['double_col'])
     backend.assert_series_equal(result, expected, check_names=False)
 
 
+@pytest.mark.only_on_backends([Pandas, PySpark])
+@pytest.mark.xfail_unsupported
+def test_elementwise_udf_mutate(backend, alltypes, df):
+    expr = alltypes.mutate(incremented=add_one(alltypes['double_col']))
+    result = expr.execute()
+
+    expected = df.assign(incremented=add_one.func(df['double_col']))
+
+    backend.assert_series_equal(result['incremented'], expected['incremented'])
+
+
+@pytest.mark.only_on_backends([Pandas, PySpark])
+@pytest.mark.xfail_unsupported
+def test_analytic_udf(backend, alltypes, df):
+    result = calc_zscore(alltypes['double_col']).execute()
+    expected = calc_zscore.func(df['double_col'])
+    backend.assert_series_equal(result, expected, check_names=False)
+
+
+@pytest.mark.only_on_backends([Pandas, PySpark])
+@pytest.mark.xfail_unsupported
+def test_analytic_udf_mutate(backend, alltypes, df):
+    expr = alltypes.mutate(zscore=calc_zscore(alltypes['double_col']))
```

Comment:
Hmm..Don't you need to define a unbounded window in order to use analytics udf?

Reply:
Where is the implicit conversion happening?

Reply:
Yes apparently `mutate` implicitly applies the analytic UDF over a window. The logic is like this:

1. A mutation is a projection
2. When doing a projection, `Projector` is used to pre-process the projection ([code](https://github.com/ibis-project/ibis/blob/3a3b46edca87cbe8ba601406a2ebb8536904d2dd/ibis/expr/api.py#L4231))
3. `Projector` calls `windowize_function` on all expressions ([code](https://github.com/ibis-project/ibis/blob/3a3b46edca87cbe8ba601406a2ebb8536904d2dd/ibis/expr/analysis.py#L766))
4. `windowize_function` applies `AnalyticOps` over an unbounded window ([code](https://github.com/ibis-project/ibis/blob/3a3b46edca87cbe8ba601406a2ebb8536904d2dd/ibis/expr/analysis.py#L713))

--

For reference this is what the expression looks like (has `WindowOp` despite not being explicitly defined)
```
expr = alltypes.mutate(zscore=calc_zscore(alltypes['double_col']))
print(expr)
```

```
ref_0
PandasTable[table]
  ...

Selection[table]
  table:
    Table: ref_0
  selections:
    Table: ref_0
    zscore = WindowOp[float64*]
      zscore = AnalyticVectorizedUDF[float64*]
        func:
          <function calc_zscore at 0x7f4b8f3c6560>
        func_args:
          double_col = Column[float64*] 'double_col' from table
            ref_0
        input_type:
          float64
        _output_type:
          float64
      <ibis.expr.window.Window object at 0x7f4b8ec6d5d0>
```

Reply:
Make sense. I remember the `windowize_function ` now

Reply:
can you put this somewhere for posterity (maybe in the projection doc-string? )